tasks/node/agent/version.go

```diff
@@ -36,63 +38,46 @@ func (t NodeAgentVersion) Dependencies() []string {
 
 // Execute - The core work within this task
 func (t NodeAgentVersion) Execute(options tasks.Options, upstream map[string]tasks.Result) tasks.Result {
-	var result tasks.Result
+	var nodeModuleVersions []NodeEnv.NodeModuleVersion
+	var ok bool
 
 	if upstream["Node/Config/Agent"].Status != tasks.Success {
-		result.Status = tasks.None
-		result.Summary = "Node agent not detected"
-		return result
-	}
-	logs, ok := upstream["Base/Log/Copy"].Payload.([]logtask.LogElement)
-	if !ok {
-		result.Status = tasks.None
-		result.Summary = "Node agent version not detected"
-		return result
+		return tasks.Result{
+			Status:  tasks.None,
+			Summary: "Node agent not detected",
+		}
 	}
-	agentVersion := getNodeVerFromLog(logs)
-	log.Debug("Agent version", agentVersion)
-	for _, nodeLog := range agentVersion {
-		if nodeLog.MatchFound == true {
-			result.Status = tasks.Info
-			result.Summary = nodeLog.AgentVersion
-			result.Payload = nodeLog.AgentVersion
-			return result
+	if upstream["Node/Env/Dependencies"].Status == tasks.Info {
+		nodeModuleVersions, ok = upstream["Node/Env/Dependencies"].Payload.([]NodeEnv.NodeModuleVersion)
+		if !ok {
+			return tasks.Result{
+				Status:  tasks.None,
+				Summary: "Node module version not detected",
+			}
 		}
 	}
-	return result
-}
 
-func getNodeVerFromLog(logs []logtask.LogElement) (agentVersion []logNodeAgentVersion) {
-	for _, logElement := range logs {
-		logAgentVersion, errSearchLogs := searchLogs(logElement)
-		var logElementNode logNodeAgentVersion
-		if errSearchLogs != nil || logAgentVersion == "" {
-			logElementNode.Logfile = logElement.FilePath + logElement.FileName
-			logElementNode.AgentVersion = ""
-			logElementNode.MatchFound = false
-		} else {
-			logElementNode.Logfile = logElement.FilePath + logElement.FileName
-			logElementNode.AgentVersion = logAgentVersion
-			logElementNode.MatchFound = true
+	agentVersion := getNodeVerFromPayload(nodeModuleVersions)
+	if agentVersion == "" {
+		return tasks.Result{
+			Status:  tasks.Warning,
+			Summary: "Node Agent Module not found for newrelic",
```

Comment:
Maybe we should add a more explicit message to the user, like:  "We were unable to find the 'newrelic' module required for the Node Agent installation. Make sure to run 'npm install newrelic' and verify that 'newrelic' is listed in your package.json"

Reply:
That's a great message. Thanks for the suggestion.